### ISSUE TITLE
Include pylint in build or test procedure

### ISSUE THREAD

**Issue by max-veit** [libAtoms/QUIP#29]
This fairly standard tool checks Python code for a number of common syntax derps and logic errors (such as undefined names and missing imports, things that wouldn't be obvious at first glance). It's no substitute for better test coverage but it catches the small errors much more quickly.

By default, the pylint tool spews lots of information about minor stylistic nonconformities as well, but maybe we can run it in errors-only mode at first. It seems there's also a way to suppress the few spurious errors it reports for quippy modules.

**Comment by gabor1** [comment]
sounds good… James?  

— Gábor

**Comment by tdaff** [comment]
I'm in favour of static checks. I'm also thinking about measuring test coverage too.

Before doing too much code cleanup @max-veit what does @jameskermode think about integrating a more up to date version of f90wrap? That might catch some errors and save duplicating effort. It also helps if we want to aim for Python 3 compatibility in the future.

**Comment by jameskermode** [comment]
Yup, no objections, but agree with @tdaff that switching quippy over to use current f90wrap to build the auto-generated modules is the highest priority task. f90wrap is already Python 3 compatible so this would be a big step forward. The only reason I haven't done it already is lack of time, so if someone would be keen to try it I would be more than happy to help out with fixing any problems that come up.
